tools/site_search.php

```diff
@@ -0,0 +1,343 @@
+            "url" => get_url_for_inbox(),
+            "desc" => _("Inbox"),
+        ];
+    }
+
+    if (!empty($wiki_url)) {
+        $jump_words["wiki"] = [
+            "url" => $wiki_url,
+            "desc" => _("Wiki"),
+        ];
+    }
+
+    if (!empty($blog_url)) {
+        $jump_words["blog"] = [
+            "url" => $wiki_url,
+            "desc" => _("Blog"),
+        ];
+    }
+
+    if (user_is_PM()) {
+        $jump_words["pm"] = [
+            "url" => "$code_url/tools/project_manager/projectmgr.php",
+            "desc" => _("Project Management"),
+        ];
+        $jump_words["rfm"] = [
+            "url" => "$code_url/tools/project_manager/remote_file_manager.php",
+            "desc" => _("Remote File Manager"),
+        ];
+    }
+
+    if (user_is_a_sitemanager()) {
+        $jump_words["sa"] = [
+            "url" => "$code_url/tools/site_admin/index.php",
+            "desc" => _("Site Administration"),
+        ];
+    }
+
+    if ($flatten_aliases) {
+        foreach ($jump_words as $key => $data) {
+            if (isset($jump_words[$key]["alias"])) {
+                $jump_words[$jump_words[$key]["alias"]] = $data;
+                unset($jump_words[$key]["alias"]);
+            }
+        }
+    }
+
+    ksort($jump_words);
+    return $jump_words;
+}
+
+function get_activity_jumps(): array
+{
+    global $code_url;
+
+    $activity_jumps = [];
+    foreach (Activities::get_all() as $activity) {
+        if ($activity instanceof Round) {
+            $activity_jumps[$activity->id] = "$code_url/tools/proofers/round.php?round_id=$activity->id";
+        } elseif ($activity instanceof Pool) {
+            $activity_jumps[$activity->id] = "$code_url/tools/pool.php?pool_id=$activity->id";
+        } elseif ($activity->id === "SR") {
+            // SR is such a special snowflake
+            $activity_jumps[$activity->id] = "$code_url/tools/post_proofers/smooth_reading.php";
+        }
+    }
+
+    return $activity_jumps;
+}
+
+function jump_search(string $query)
+{
+    $query = strtolower($query);
+    $jump_words = get_jump_words(true);
+    if (isset($jump_words[$query])) {
+        metarefresh(0, $jump_words[$query]["url"], "", "", true);
+    }
+}
+
+function smart_search(string $query)
+{
+    global $code_url;
+
+    // is it a valid projectID?
+    try {
+        new Project($query);
+        metarefresh(0, "$code_url/project.php?id=$query");
+    } catch (NonexistentProjectException $e) {
+        // pass
+    }
+
+    // is it a valid username?
+    try {
+        $user = new User($query);
+        metarefresh(0, "$code_url/stats/members/mdetail.php?id=$user->u_id");
+    } catch (NonexistentUserException $e) {
+        // pass
+    }
+
+    // is it an Activity?
+    $activity_jumps = get_activity_jumps();
+    if (isset($activity_jumps[strtoupper($query)])) {
+        metarefresh(0, $activity_jumps[strtoupper($query)]);
+    }
+}
+
+function prefix_search(string $query)
+{
+    $matches = [];
```

Comment:
Again, I'm probably being fussy here, but we have a mild case of 'arrow code' here. We could avoid the nested indentation with 
```
if (preg_match('/^(\w+):\s*(.*)$/', $query, $matches) === false) return;
...
$data = ...
if ($data == null) return;
...
$query = ...
metarefresh(...);
```